ciao-controller/internal/datastore/sqlite3db.go

```diff
@@ -1139,6 +1155,46 @@ func (ds *sqliteDB) updateWorkload(w types.Workload) error {
 	return nil
 }
 
+func (ds *sqliteDB) deleteWorkload(ID string) error {
+	db := ds.getTableDB("workload_template")
+
+	ds.dbLock.Lock()
+	defer ds.dbLock.Unlock()
+
+	tx, err := db.Begin()
+	if err != nil {
+		return err
+	}
+
+	err = ds.deleteWorkloadDefault(tx, ID)
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+
+	err = ds.deleteWorkloadStorage(tx, ID)
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+
+	filename := fmt.Sprintf("%s_config.yaml", ID)
+	path := fmt.Sprintf("%s/%s", ds.workloadsPath, filename)
```

Comment:
Would it not be better to delete the file right at the end.  If the "DELETE FROM workload_template WHERE id = ?" fails, the transaction will be rolled back but the _config.yaml file will be gone.